nodejs/eks/nodegroup.ts

```diff
@@ -704,6 +764,8 @@ export function createManagedNodeGroup(name: string, args: ManagedNodeGroupOptio
     const nodeGroup = new aws.eks.NodeGroup(name, {
         ...nodeGroupArgs,
         clusterName: args.clusterName || core.cluster.name,
+        nodeGroupName: args.nodeGroupName || name,
+        nodeRoleArn: roleArn,
```

Comment:
This is also a breaking change I assume?  I can't tell exactly why we are doing this to understand exactly what the impact is.  What will happen to existing users?  Why do we believe we need to change this?

Reply:
- `nodeGroupName`: on the init release, this was a user provided arg that could only be set via `aws.eks.NodeGroupArgs` directly. we've now pulled it into `ManagedNodeGroupOptions` (a Pick of `aws.eks.NodeGroupArgs`) to allow its optionality of configuration, and defaulting if not provided. Existing clusters should not see anything differently if they already provided this prop as it exists under the same name.
- `nodeRoleArn`: same goes for this prop as `nodeGroupName` above - it was only able to be set via `aws.eks.NodeGroupArgs` and we pulled it into `ManagedNodeGroupOptions` for optionality and flexibility of passing a role or roleArn to the nodegroup by the user:
  - per https://github.com/pulumi/pulumi-eks/issues/293#issuecomment-568085092
  > I don't like the inconsistency in what is passed to the two classes but that's not our problem to solve right now. One takes an ARN, the other takes a role.